src/app/site/checkout/checkout-page/checkout-page.component.ts

```diff
@@ -85,28 +95,86 @@ export class CheckoutPageComponent implements OnInit {
       });
 
     this.checkoutService.isValid$.subscribe(r => this.disabled = (!r && !this.soldOuts));
+    if (!this.authService.userDetails.userId) {
+      this.showEarnPointLabel = false;
+      this.earnedLoyaltyPoint = 0;
+    } else {
+      this.showEarnPointLabel = true;
+      this.checkoutService.loyaltyGroups.subscribe(data => this.loyaltyGroups = data);
+      this.checkoutService.addPointArray.subscribe(data => this.addPointArray = data[0].add_point);
+    }
+
+    // this.checkoutService.setPaymentType(this.paymentType.cash);
+    this.checkoutService.setPaymentType(this.paymentType[this.selectedPaymentType]);
   }
 
   changePaymentType(data) {
     this.usedBalance = 0;
     this.usedLoyaltyPoint = 0;
+    this.selectedPaymentType = data;
 
     switch (data) {
+      case this.paymentType.cash: {
+        this.checkoutService.setPaymentType(data);
+      }
+        break;
       case this.paymentType.balance: {
         this.usedBalance = this.balanceValue;
         this.checkoutService.setPaymentType(data);
       }
         break;
       case this.paymentType.loyaltyPoint: {
-        this.usedLoyaltyPoint = this.loyaltyPoint;
+        this.usedLoyaltyPoint = this.loyaltyPoint * this.loyaltyValue;
         this.checkoutService.setPaymentType(data);
       }
         break;
     }
+
+    this.calculateEarnPoint();
   }
 
-  showDiscountLabel(data) {
+  setCostLabel(data) {
     this.showCostLabel = data;
+    this.calculateEarnPoint();
+  }
+
+  calculateEarnPoint() {
+    let scoreArray;
+    let maxScore;
+    let customer_loyaltyGroup;
+    let valid_loyaltyGroups;
+
+    if (!this.authService.userDetails.userId) {
+      this.earnedLoyaltyPoint = 0;
+      this.showEarnPointLabel = false;
+      return;
+    }
+
+    if (this.selectedPaymentType === 2)
```

Comment:
highly recommended to compare selectedPaymentType with enum (2's meaning is not clear)